.github/workflows/integrate.yml

```diff
@@ -0,0 +1,71 @@
+name: Run PHPUnit and PHPCS
+
+on: [push]
+
+jobs:
+  test:
+    name: WP ${{ matrix.wordpress }} on PHP ${{ matrix.php }}
+    # Ubuntu-20.x includes MySQL 8.0, which causes `caching_sha2_password` issues with PHP < 7.4
+    # https://www.php.net/manual/en/mysqli.requirements.php
+    # TODO: change to ubuntu-latest when we no longer support PHP < 7.4
+    runs-on: ubuntu-18.04
+    continue-on-error: ${{ matrix.allowed_failure }}
+
+    env:
+      WP_VERSION: ${{ matrix.wordpress }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        php: ["5.6", "7.0", "7.1", "7.2", "7.3", "7.4", "8.0"]
+        wordpress: ["5.5", "5.6", "5.7"]
+        allowed_failure: [false]
+        # https://make.wordpress.org/core/2020/11/23/wordpress-and-php-8-0/
+        exclude:
+          - php: "8.0"
+            wordpress: "5.5"
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up PHP ${{ matrix.php }}
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ matrix.php }}
+          coverage: pcov
+          # https://make.wordpress.org/hosting/handbook/handbook/server-environment/#php-extensions
+          extensions: curl, dom, exif, fileinfo, hash, json, mbstring, mysqli, openssl, pcre, imagick, xml, zip
+
+      - name: Install Composer dependencies (PHP < 8.0 )
+        if: ${{ matrix.php < 8.0 }}
+        uses: ramsey/composer-install@v1
+
+      - name: Install Composer dependencies (PHP >= 8.0)
+        if: ${{ matrix.php >= 8.0 }}
+        uses: ramsey/composer-install@v1
+        with:
+          composer-options: --ignore-platform-reqs
```

Comment:
By moving the composer options into the matrix, then only one step needs to be defined here for installing Composer dependencies.

See https://github.com/Parsely/wp-parsely/blob/develop/.github/workflows/integration-tests.yml#L65-L68 and https://github.com/Parsely/wp-parsely/blob/develop/.github/workflows/integration-tests.yml#L30 (for the PHP [5.6...7.4] matrix, the `composer-options` is not defined, therefore as an empty string value).